src/components/banner/BannerErrorState.tsx

```diff
@@ -0,0 +1,173 @@
+import { useScaleAnimation } from "../../hooks";
+import { VSpacer } from "../spacer";
+import { Body } from "../typography";
+import { ButtonLink } from "../buttons";
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: "row",
+    alignItems: "center",
+    alignContent: "center",
+    borderRadius: IOBannerRadius,
+    borderCurve: "continuous",
+    padding: IOBannerBigSpacing
+  }
+});
+
+/* Component Types */
+
+type BaseBannerErrorStateProps = WithTestID<{
+  icon?: IOIcons;
+  label: string;
+  viewRef?: RefObject<View>;
+  // A11y related props
+  accessibilityLabel?: string;
+  accessibilityHint?: string;
+}>;
+
+type BannerErrorStateActionProps =
+  | {
+      actionText?: string;
+      onPress: (event: GestureResponderEvent) => void;
+      accessibilityRole?: never;
+    }
+  | {
+      actionText?: never;
+      onPress?: never;
+      accessibilityRole?: AccessibilityRole;
+    };
+
+export type BannerErrorStateProps = BaseBannerErrorStateProps &
+  BannerErrorStateActionProps;
+
+/**
+ ** TODO: Move it to the `io-app-design-system` package
+ */
+export const BannerErrorState = ({
+  viewRef,
+  icon = "warningFilled",
+  label,
+  actionText,
+  onPress,
+  accessibilityHint,
+  accessibilityLabel,
+  testID
+}: BannerErrorStateProps) => {
+  const { onPressIn, onPressOut, scaleAnimatedStyle } =
+    useScaleAnimation("medium");
+
+  const theme = useIOTheme();
+  const { themeType } = useIOThemeContext();
+
+  /* Styles */
+  const foregroundColor: IOColors = theme["textBody-tertiary"];
+  const backgroundColor: IOColors = "grey-50";
+
+  const dynamicContainerStyles: ViewStyle = {
+    backgroundColor:
+      themeType === "dark"
+        ? hexToRgba(IOColors[backgroundColor], 0.1)
+        : IOColors[backgroundColor]
+  };
+
+  const renderMainBlock = () => (
+    <>
+      <View
+        style={{ flex: 1, alignItems: "center" }}
+        accessible={true}
+        // A11y related props
+        accessibilityLabel={accessibilityLabel}
+        accessibilityHint={accessibilityHint}
+        accessibilityRole={actionText !== undefined ? "button" : undefined}
+      >
+        {icon && (
+          <>
+            <Icon name={icon} size={24} color={foregroundColor} />
+            <VSpacer size={8} />
+          </>
+        )}
+        {label && (
+          <>
+            <Body color={foregroundColor} textStyle={{ textAlign: "center" }}>
+              {label}
+            </Body>
+            {actionText && <VSpacer size={8} />}
+          </>
+        )}
+        {actionText && (
+          /* Disable pointer events to avoid
+            pressed state on the button */
+          <View
+            pointerEvents="none"
+            accessibilityElementsHidden
+            importantForAccessibility="no-hide-descendants"
+          >
+            <VSpacer size={4} />
+            <ButtonLink color="primary" onPress={onPress} label={actionText} />
+          </View>
+        )}
+      </View>
+    </>
```

Comment:
Since a fragment with only one child is redundant, what do you think to remove it?
```suggestion
     <View
      style={{ flex: 1, alignItems: "center" }}
      accessible={true}
      // A11y related props
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityRole={actionText !== undefined ? "button" : undefined}
    >
      {icon && (
        <>
          <Icon name={icon} size={24} color={foregroundColor} />
          <VSpacer size={8} />
        </>
      )}
      {label && (
        <>
          <Body color={foregroundColor} textStyle={{ textAlign: "center" }}>
            {label}
          </Body>
          {actionText && <VSpacer size={8} />}
        </>
      )}
      {actionText && (
        /* Disable pointer events to avoid
            pressed state on the button */
        <View
          pointerEvents="none"
          accessibilityElementsHidden
          importantForAccessibility="no-hide-descendants"
        >
          <VSpacer size={4} />
          <ButtonLink color="primary" onPress={onPress} label={actionText} />
        </View>
      )}
    </View>
```